ios/FluentUI/Bottom Commanding/BottomCommandingController.swift

```diff
@@ -0,0 +1,586 @@
+        }
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        guard previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass else {
+            return
+        }
+
+        // On a horizontal size class change the top level sheet / bar surfaces get recreated,
+        // but the item views, containers and bindings persist and are rearranged during the individual setup functions.
+        if let bottomSheetController = bottomSheetController {
+            bottomSheetController.willMove(toParent: nil)
+            bottomSheetController.removeFromParent()
+            bottomSheetController.view.removeFromSuperview()
+        }
+        bottomSheetController = nil
+        bottomBarView?.removeFromSuperview()
+        bottomBarView = nil
+
+        if traitCollection.horizontalSizeClass == .regular {
+            setupBottomBarLayout()
+        } else {
+            setupBottomSheetLayout()
+        }
+    }
+
+    public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        if presentedViewController != nil {
+            dismiss(animated: false)
+        }
+    }
+
+    private func setupBottomBarLayout() {
+        NSLayoutConstraint.activate(heroCommandWidthConstraints)
+        heroCommandStack.distribution = .equalSpacing
+
+        let commandContainer = UIStackView()
+        commandContainer.translatesAutoresizingMaskIntoConstraints = false
+        commandContainer.addArrangedSubview(heroCommandStack)
+        commandContainer.addArrangedSubview(moreButtonView)
+
+        let bottomBarView = makeBottomBarByEmbedding(contentView: commandContainer)
+        bottomBarView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(bottomBarView)
+
+        NSLayoutConstraint.activate([
+            bottomBarView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            bottomBarView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -Constants.BottomBar.bottomOffset)
+        ])
+
+        self.bottomBarView = bottomBarView
+        updateExpandability()
+    }
+
+    private func setupBottomSheetLayout() {
+        NSLayoutConstraint.deactivate(heroCommandWidthConstraints)
+        heroCommandStack.distribution = .fillEqually
+
+        let commandStackContainer = UIView()
+        commandStackContainer.addSubview(heroCommandStack)
+
+        let sheetController = BottomSheetController(contentView: makeBottomSheetContent(headerView: commandStackContainer, expandedContentView: tableView))
+        sheetController.hostedScrollView = tableView
+        sheetController.collapsedContentHeight = bottomSheetHeroStackHeight
+        sheetController.expandedHeightFraction = Constants.BottomSheet.expandedFraction
```

Comment:
is there a way for clients to put expanded height based on number of cells they want to expand? it's not always going to be clear cut of .5 or 1

Reply:
might as well have respectsPreferredContentSize case in the demo app for demonstration and testability

Reply:
More intelligent scaling is coming in the future. This is more of a placeholder for now. In the future we'll be deciding between two expansion states and one, based on how much space the expanded content takes up.